ompi/mca/mtl/ofi/configure.m4

```diff
@@ -24,10 +24,15 @@ AC_DEFUN([MCA_ompi_mtl_ofi_POST_CONFIG], [
 #                    [action-if-cant-compile])
 # ------------------------------------------------
 AC_DEFUN([MCA_ompi_mtl_ofi_CONFIG],[
-    OPAL_VAR_SCOPE_PUSH([mtl_ofi_happy])
+    OPAL_VAR_SCOPE_PUSH([mtl_ofi_happy common_ofi_compile_mode mtl_ofi_compile_mode])
 
     AC_CONFIG_FILES([ompi/mca/mtl/ofi/Makefile])
 
+    MCA_COMPONENT_COMPILE_MODE("opal", "common", "ofi", common_ofi_compile_mode)
```

Comment:
I don't think we want to call `MCA_COMPONENT_COMPILE_MODE` as it exists today from here.  Two reasons: first, it's going to print out an `AC_MSG_CHECKING` that will be very confusing.  Second, this code almost certainly will be copied around (you already call out UCX, but any component that depends on a common component needs such a check), and there's an implicit assumption that neither the calling component nor the common component override the `MCA_COMPONENT_COMPILE_MODE` option.

I think both problems are solvable relatively easily, but won't have time to post a patch until tomorrow night.